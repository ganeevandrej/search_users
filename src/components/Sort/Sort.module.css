.sort {
    width: 70%;
    margin: auto;
    padding-top: 20px;
    padding-bottom: 10px;
}

.wrapper {
    width: 20%;
}