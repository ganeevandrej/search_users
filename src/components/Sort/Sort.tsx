import React from "react";
import {Select, SelectChangeEvent, MenuItem} from "@mui/material";
import style from "./Sort.module.css";
import {SortRotation} from "../../services/constants";
import {PropsSort} from "../../services/interfaces";

export const Sort: React.FC<PropsSort> = ({value, onChange}): React.JSX.Element => {
    const handleChange = (event: SelectChangeEvent<typeof value>)  => {
        onChange(event.target.value);
    }

    return (
        <div className={style.sort}>
            <div className={style.wrapper}>
                <Select
                    data-testid="select"
                    size="small"
                    value={value}
                    onChange={handleChange}
                    defaultValue={SortRotation.DESC}>
                    <MenuItem data-testid="asc" value={SortRotation.ASC}>по возрастанию</MenuItem>
                    <MenuItem data-testid="desc" value={SortRotation.DESC}>по убыванию</MenuItem>
                </Select>
            </div>
        </div>
    );
}