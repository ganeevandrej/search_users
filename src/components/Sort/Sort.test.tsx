import React from "react";
import {screen, render} from "@testing-library/react";
import {Sort} from "./Sort";
import userEvent from '@testing-library/user-event';

describe("Testing the Sort component", () => {
    test("snapshot of open select", async () => {
        const onChange = jest.fn();
        render(<Sort value="desc" onChange={onChange}  />);
        await userEvent.dblClick(screen.getByRole("button"));
        const listBox = screen.getByRole('listbox');
        expect(listBox).toMatchSnapshot();
    })
})