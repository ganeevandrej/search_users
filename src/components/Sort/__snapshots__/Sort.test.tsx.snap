// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Sort component snapshot of open select 1`] = `
<ul
  class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
  role="listbox"
  tabindex="-1"
>
  <li
    aria-selected="false"
    class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
    data-testid="asc"
    data-value="asc"
    role="option"
    tabindex="-1"
  >
    по возрастанию
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </li>
  <li
    aria-selected="true"
    class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters Mui-selected MuiMenuItem-root MuiMenuItem-gutters Mui-selected css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
    data-testid="desc"
    data-value="desc"
    role="option"
    tabindex="0"
  >
    по убыванию
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </li>
</ul>
`;
